build-aux/m4/l_filesystem.m4

```diff
@@ -0,0 +1,47 @@
+dnl Copyright (c) 2022 The Bitcoin Core developers
+dnl Distributed under the MIT software license, see the accompanying
+dnl file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+# GCC 9.1 and earlier requires -lstdc++fs
+# Clang 9.0 (libc++) and earlier requires -lc++fs
```

Comment:
def4c1704d27754398375880d9a0ba8f31a0464d

This comment makes me think that using `CXX='clang++-9 -stdlib=libc++'` ends with linking `-lc++fs`. That is not the case. Tested on tested on Linux Mint 20.2 (x86_64) with [clang-8](https://packages.ubuntu.com/focal/clang-8) and [clang-9](https://packages.ubuntu.com/focal/clang-9).

See https://releases.llvm.org/9.0.0/projects/libcxx/docs/UsingLibcxx.html#using-filesystem

Maybe
```suggestion
# Prior to LLVM 9.0, libc++ provides the implementation of the filesystem library in a separate static library,
# and using of <filesystem> requires to link -lc++fs.
```
or something similar with better wording?

Reply:
Similar concerns are about gcc comment above as according to https://gcc.gnu.org/gcc-9/changes.html gcc-9 does not require `-lstdc++fs`.

Reply:
I don't think we need to copy paste all of that in here. We can just change it to: 
```suggestion
# Clang 8.0 (libc++) and earlier requires -lc++fs
```

Reply:
Hmm, unexpected:
```
$ ./configure CC=gcc-8 CXX=g++-8
...
checking whether std::filesystem can be used without link library... yes
...
```

Reply:
Addressed both.

Reply:
> Hmm, unexpected:

If you've got other GCC's installed, it's likely not using version 8 of the standard library. Have a look at the verbose compiler output.